### PR TITLE
Improve ngrok support

### PR DESCRIPTION
**Pull request by markmetcalfe**
Currently there is no guide on how to get docker-dev working with ngrok. This patch adds documentation, along with a new 'tngrok' command which simplifies the command required to get ngrok up and running with docker-dev. This patch also includes a config.php snippet for automatically setting the correct wwwroot.

### CODE REVIEW

**Review by simoncoggins — APPROVED** [review]
I haven't tested but from reading the code this looks good to me.

**Comment by markmetcalfe** [comment]
I'll give this a test quickly with the old setup. Also might be worth seeing if we can get the paid version of ngrok available for us, then it might be worth adding another tngrok argument for specifying the static URL

File: nginx/entrypoint.sh [hunk 1]
```diff
@@ -2,23 +2,23 @@
 
 echo "Using data folder: $REMOTE_DATA"
 
-cd $REMOTE_DATA
+cd $REMOTE_DATA || return;
 
 echo "Creating data folders..."
 
-versions=("22" "24" "25" "26" "27" "29" "9" "10" "11" "12" "13")
+versions=("22" "24" "25" "26" "27" "29" "9" "10" "11" "12" "13" "14")
```
Comment (derschatta) [inline nginx/entrypoint.sh:9]: we need 15 now too

**Review by derschatta — CHANGES_REQUESTED** [review]
tested it successfully with some modifications, see comments

File: nginx/entrypoint.sh (hunk 1 above)
Comment (markmetcalfe) [reply nginx/entrypoint.sh]: fixed

File: README.md [hunk 2]
```diff
@@ -610,6 +616,15 @@ mutagen sync monitor totara
 
 You can customise the docker compose configurations simply adding your own `.yml` or `.yaml` compose files into the `custom` folder. Any containers or other options you have will automatically override any existing default container options.
 
+### Ngrok
+
+__[Ngrok](https://ngrok.com/)__ is a useful tool for making your local totara site accessible publicly on the web.
```
[the remark this answers is not in the document]
Comment (markmetcalfe) [reply README.md:622]: done

File: README.md [hunk 3]
```diff
@@ -288,7 +289,12 @@ $CFG->dboptions = array(
 );
 
 // This detects and sets the wwwroot dynamically so you don't have to manually change it
-if (!empty($_SERVER['SERVER_NAME']) && !empty($_SERVER['REQUEST_SCHEME'])) {
+if (!empty($_SERVER['HTTP_X_ORIGINAL_HOST'])) {
```
[the remark this answers is not in the document]
Comment (markmetcalfe) [reply README.md]: fixed

**Comment by markmetcalfe** [comment]
@derschatta Fixed your comments. Could you approve this?

**Review by derschatta — CHANGES_REQUESTED** [review]
just the one comment. rest looks good and works well

File: bin/tngrok [hunk 4]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (markmetcalfe) [reply bin/tngrok:1]: @derschatta Fixed

**Review by derschatta — APPROVED** [review]

**Pull request merged**